packages/core/src/animation/internal/AnimationCurveLayerOwner.ts

```diff
@@ -1,12 +1,29 @@
+import { IAnimationCurveCalculator } from "../animationCurve/interfaces/IAnimationCurveCalculator";
 import { KeyframeValueType } from "../Keyframe";
 import { AnimationCurveOwner } from "./animationCurveOwner/AnimationCurveOwner";
 
 /**
  * @internal
  */
-export class AnimationCurveLayerOwner {
+export class AnimationCurveLayerOwner<V extends KeyframeValueType> {
   crossSrcCurveIndex: number;
   crossDestCurveIndex: number;
   crossCurveMark: number = 0;
-  curveOwner: AnimationCurveOwner<KeyframeValueType>;
+  curveOwner: AnimationCurveOwner<V>;
+  lastValue: V;
+
+  private _cureType: IAnimationCurveCalculator<V>;
+
+  constructor(cureType: IAnimationCurveCalculator<V>) {
+    this._cureType = cureType;
+  }
+
+  saveLastValue(): void {
+    const owner = this.curveOwner;
+    if (this._cureType._isReferenceType) {
+      owner.getCurrentValue(this.lastValue);
+    } else {
+      this.lastValue = owner.getCurrentValue(this.lastValue);
+    }
+  }
```

Comment:
Adjust _curveType to @internal  in AnimationCurveOwner, then simple code like this:
```ts
/**
 * @internal
 */
export class AnimationCurveLayerOwner {
  crossSrcCurveIndex: number;
  crossDestCurveIndex: number;
  crossCurveMark: number = 0;
  curveOwner: AnimationCurveOwner<KeyframeValueType>;
  lastValue: KeyframeValueType;

  saveFinalValue(): void {
    const owner = this.curveOwner;
    if (owner._cureType._isReferenceType) {
      owner.getCurrentValue(this.lastValue);
    } else {
      this.lastValue = owner.getCurrentValue(this.lastValue);
    }
  }
}
```